### PR TITLE
Fix for issues #19

### PR DESCRIPTION
**Pull request by Raistlfiren**

### CODE REVIEW

**Comment by xiaohutai** [comment]
Now, it takes the `sort` -> `order`, but it will give an ambiguous `id` error. 

```
An exception occurred while executing

SELECT COUNT(*) as total
FROM bolt_pages pages
LEFT JOIN bolt_relations incomingrelation
ON pages.id = incomingrelation.to_id
AND incomingrelation.to_contenttype='pages'
LEFT JOIN bolt_taxonomy groups
ON pages.id = groups.content_id
AND groups.contenttype='pages'
AND groups.taxonomytype='groups'
ORDER BY id ASC:

SQLSTATE[HY000]: General error: 1 ambiguous column name: id
```

I'm not sure how the query builder works, but I think everything that is passed through `sort` or `order`, needs to be handled differently and prefixed with the content type or taxonomy.

File: src/Action/SearchAction.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function handle($contentType = null, Request $request, ParameterCollectio
 
         if ($contentType === null) {
             //Remove search order off of id, since ambigious...
-            unset($queryParameters['order']);
+            unset($queryParameters['sort']);
```
Comment (xiaohutai) [inline src/Action/SearchAction.php]: This should be `order` I think, as this is the one that is fed to Bolt.

**Comment by xiaohutai** [comment]
OK, I've did some more testing.
As far as I know two things currently break:
- sorting by `id` (or not specifiying a `&sort=` defaults to `id`).
- searching through all contenttypes (`/json/search?q=`) returns nothing

**Comment by Raistlfiren** [comment]
Yeah, changing this - https://github.com/xiaohutai/jsonapi/blob/master/src/Storage/Query/Handler/PagingHandler.php#L61 to reset orderBy fixes the problem with ambiguous columns. 

The searching through all content types is a problem with the search implementation as noted here - https://github.com/bolt/bolt/issues/5391

I am trying to create a fix for it right now.

**Comment by xiaohutai** [comment]
@Raistlfiren Thanks for the update. Again, no stress.

Sorting search results is a bit problematic I think. Since there's an implicit sorting by _relevancy_. So these results should - by default - never be sorted by `id`, but when sorting by `id` is explicitly requested, then do sort it.

Hmm, and someone needs to re-do the Bolt 3 search :sweat_smile:

**Comment by Raistlfiren** [comment]
This problem should now be corrected. Search should also be fixed. Thank you for pinging me @xiaohutai

**Comment by xiaohutai** [comment]
I can't test at the moment. Looking at the code, it looks good to me.
Many thanks, once again @Raistlfiren

**Pull request merged**